pontoon/sync/core.py

```diff
@@ -165,6 +165,27 @@ def update_translated_resources(db_project, vcs_project, locale):
                 translatedresource.calculate_stats()
 
 
+def update_translated_resources_no_files(db_project, locale, changed_resources):
+    """
+    Create/update TranslatedResource entries if files aren't available. This tipically happens when
+    originals change and translations don't, so we don't pull locale repositories.
+    """
+    for resource in changed_resources:
+        # We can only update asymmetric (monolingual) TranslatedResources. For bilingual files we
+        # only create TranslatedResources if the file is present in the repository for the locale,
+        # which we cannot check without files.
+        if not resource.is_asymmetric:
+            log.error(
+                'Unable to calculate stats for asymmetric resource: {resource}.'.format(resource)
+            )
+            continue
```

Comment:
This should never happen, because bilingual files need to be updated for all locales (merged), when the source change.

Still, maybe we should throw an exception here instead, so that we notice the issue?

Reply:
What would happen on next sync if we do?

Reply:
Regardless of raising an exception, we no longer have the information about the project changes in the following sync. That's [bug 1383271](https://bugzilla.mozilla.org/show_bug.cgi?id=1383271).

Reply:
I'd not error out here for now. Also, I'm not 100% confident that we should restrict ourselves to have projects with only monolingual or only bilingual files.

Reply:
We have projects with a mix of monolingual or bilingual files (e.g. Fundraising). They should work fine with this code.

If the source .lang file changes, all locales will be updated in the repository, so we'll have files for all locales and use the current `update_translated_resources` method.

If only the source .properties file changes, we'll use `update_translated_resources_no_files` for locales that don't have any locale changes.